WordPress/Classes/Services/Store.swift

```diff
@@ -22,53 +22,155 @@ class StoreKitTransactionObserver: NSObject, SKPaymentTransactionObserver {
 // stored property since it's a generic class.
 struct StoreKitCoordinator {
     static let instance = StoreCoordinator(store: StoreKitStore())
+    
+    static let TransactionDidFinishNotification = "StoreCoordinatorTransactionDidFinishNotification"
+    static let TransactionDidFailNotification   = "StoreCoordinatorTransactionDidFailNotification"
+    static let NotificationProductIdentifierKey = "StoreCoordinatorNotificationProductIdentifierKey"
 }
 
+typealias PendingPayment = (planID: PlanID, productID: String, siteID: Int)
+
+/// StoreCoordinator coordinates purchasing of products, processing of transactions, and
+/// verification of purchases between the App Store (StoreKit) and WordPress.com.
+///
+/// Users of this class can simply attempt a purchase using `purchasePlan(_:product:forSite)`,
+/// and the coordinator will post a notification on a successful or failed purchase:
+///
+/// - `StoreKitCoordinator.TransactionDidFinishNotification` on success.
+///   The notification's `userInfo` will contain the productID of the purchased
+///   product under `StoreKitCoordinator.NotificationProductIdentifierKey`.
+/// - `StoreKitCoordinator.TransactionDidFailNotification` on failure.
+///   The notification's `userInfo` will also contain the productID of the attempted
+///   purchased product, as well as a localized error message under `NSUnderlyingErrorKey`.
 class StoreCoordinator<S: Store> {
-    let store: S
-    var pendingPayment: (plan: Plan, siteID: Int)? = nil
+    private let store: S
+    
+    var pendingPayment: PendingPayment? {
```

Comment:
I saw you made this `private` then reverted it. Which made me wonder if we can keep it private and do tests differently. Perhaps with a mock store, you just `purchasePlan` if you want to have a pending payment